http/agent.go

```diff
@@ -308,3 +319,221 @@ func (a *Agent) PostToWriter(w io.Writer, url string, postData []byte) error {
 	}
 	return a.readResponse(resp, w)
 }
+
+// GetRequestGroup behaves like agent.SendGetRequest() but takes a group of URLs
+// and performs the requests in parallel. The number of simultaneous requests is
+// controlled by options.MaxParallel.
+func (a *Agent) GetRequestGroup(urls []string) ([]*http.Response, []error) {
+	t := throttler.New(int(a.options.MaxParallel), len(urls))
+	ret := []*http.Response{}
+	errs := []error{}
+	m := sync.Mutex{}
+	for _, u := range urls {
+		url := u
+		func(url string) {
+			//nolint: bodyclose // We don't close here as we're returning the response
+			resp, err := a.AgentImplementation.SendGetRequest(a.Client(), url)
+
+			m.Lock()
+			errs = append(errs, err)
+			//nolint: bodyclose // We don't close here as we're returning the response
+			ret = append(ret, resp)
+			m.Unlock()
+
+			t.Done(err)
+		}(url)
+		t.Throttle()
+	}
+
+	return ret, errs
+}
+
+// PostRequestGroup behaves like agent.Post() but takes a group of URLs and performs the
+// requests in parallel. The number of simultaneous requests is controlled by
+// options.MaxParallel.
+//
+// The list of URLs and postData byte arrays are expected to be of equal length.
+// If postData has less elements than the url list, those urls without a corresponding
+// postData array will return an error.
```

Comment:
What's the reason for not failing as early as possible, given that the function will eventually fail anyways?

Reply:
What I was trying to do was to do all requests that were plausible but that introduced an inconsistent behavior. I took your advice here and now the function will fail early if URLs and postdata don't line up. I also updated the test to check for it.